syz-fuzzer/fuzzer.go

```diff
@@ -236,27 +236,40 @@ func (tool *FuzzerTool) handleConn() {
 	}
 }
 
-func (tool *FuzzerTool) diffMaxSignal(info *flatrpc.ProgInfo, mask signal.Signal, maskCall int) {
+func (tool *FuzzerTool) diffMaxSignal(info *flatrpc.ProgInfo, mask signal.Signal, maskCall int, allSignal []int32) {
 	tool.signalMu.RLock()
 	defer tool.signalMu.RUnlock()
-	diffMaxSignal(info, tool.maxSignal, mask, maskCall)
+	diffMaxSignal(info, tool.maxSignal, mask, maskCall, allSignal)
 }
 
-func diffMaxSignal(info *flatrpc.ProgInfo, max, mask signal.Signal, maskCall int) {
+func diffMaxSignal(info *flatrpc.ProgInfo, max, mask signal.Signal, maskCall int, allSignal []int32) {
+	allExtra := false
+	var allMask uint64
+	for _, c := range allSignal {
+		if c == -1 {
+			allExtra = true
+		} else {
+			if c < 0 || c >= 64 {
```

Comment:
This puts a real restriction on the maximum allowed number of calls in a program. Earlier `prog.MaxCalls` was mostly a recommendation, now there's a real limit (64).

Should we update the comments and add some validation throughout the `prog/` code?
Or would it be better to just rewrite this code to call diffCallSignal() from the loop directly, without any unnecessary use of the bitmask?

Reply:
Done. PTAL.